### PR TITLE
mach: avoid unnecessary raise

### PR DESCRIPTION
**Pull request by xu-cheng**

### CODE REVIEW

File: Library/Homebrew/os/mac/mach.rb [hunk 1]
```diff
@@ -80,7 +80,7 @@ def mach_data
       when 0x7f454c46 # ELF
         mach_data << { :arch => :x86_64, :type => :executable }
       else
-        raise "Not a Mach-O binary."
+        return []
```
Comment (UniqMartin) [inline Library/Homebrew/os/mac/mach.rb]: Doesn't the `return` circumvent the caching that happens in this method (it starts with `@mach_data ||= begin`)?

Comment (xu-cheng) [inline Library/Homebrew/os/mac/mach.rb]: Updated to address comment.

File: Library/Homebrew/os/mac/mach.rb [hunk 2]
```diff
@@ -101,9 +99,8 @@ def mach_data
 
         mach_data << { :arch => arch, :type => type }
       end
+
       mach_data
-    rescue
-      []
```
Comment (UniqMartin) [inline Library/Homebrew/os/mac/mach.rb]: I think this `rescue` has to stay. It not only deals with the `"Not a Mach-O binary."` exception, but also with calling `unpack` on `nil`. Just try `Pathname.new(<file>).mach_data` on a file of length zero.

**Comment by UniqMartin** [comment]
Can you elaborate on the explanation “Otherwise, it could cause trouble when running `brew install` on debug mode.” from the commit message? How exactly can this cause trouble?

**Comment by xu-cheng** [comment]
It happens when I try to use `Pathname#dylib?` in formula `install`. In this case, running install with debug will invoke debrew in this raise. 

FYI, this is where I working on portable Ruby interpreter. So I used several private API. The project is in https://github.com/xu-cheng/homebrew-portable-ruby

File: Library/Homebrew/os/mac/mach.rb (hunk 2 above)
Comment (xu-cheng) [inline Library/Homebrew/os/mac/mach.rb]: Good catch on that, I now leave the `rescue` as it is but remove the unnecessary `"Not a Mach-O binary."` exception.

Comment (UniqMartin) [inline Library/Homebrew/os/mac/mach.rb]: :+1:

**Comment by UniqMartin** [comment]
> FYI, this is where I working on portable Ruby interpreter. So I used several private API. The project is in https://github.com/xu-cheng/homebrew-portable-ruby

This is interesting! Thanks for sharing!

**Pull request closed without merging**